src/python/WMCore/Services/Rucio/Rucio.py

```diff
@@ -257,12 +258,43 @@ def getReplicaInfoForBlocks(self, **kwargs):
 
         return result
 
-    def getPFN(self, nodes=None, lfns=None, destination=None, protocol='srmv2', custodial='n'):
+    def getPFN(self, nodes=None, lfns=None, destination=None, protocol=None, custodial='n'):
         """
         Copy of the method from the PhEDEx service class
         Used by CRABServer
         """
-        raise NotImplementedError("Apparently not available in the Rucio client as well")
+        nodes = nodes or []
+        lfns = lfns or []
+
+        if len(nodes) != len(lfns):
+            msg = "Node lists should have the same lenght as the lfn one."
+            raise WMRucioException(msg)
+
+        response = {}
+        for node in list(set(nodes)):
+            good_lfns = [x[1] for x  in zip(nodes, lfns) if x[0] == node]
+
+            try:
+                if protocol:
+                    pfns = self.cli.lfns2pfns(node, good_lfns, scheme=protocol)
```

Comment:
Here it is the doc (https://rucio.readthedocs.io/en/latest/api/rse.html#rucio.client.rseclient.RSEClient.lfns2pfns), not very detailed though.

The reason for `scheme=None` is that looks like working for all the RSEs, while specifying one forces the request of only that single protocol even if the RSE has more than one. Then I don't know the priorities of one protocol over an other in case of `scheme=None`, that would be interesting to understand too.

Reply:
My question was more towards simplifying this code, like instead of:
```
                if protocol:
                    pfns = self.cli.lfns2pfns(node, good_lfns, scheme=protocol)
                else:
                    pfns = self.cli.lfns2pfns(node, good_lfns)
```
just call it like
```
pfns = self.cli.lfns2pfns(node, good_lfns, scheme=protocol)
```

Reply:
oh right, absolutely non-sense there. I'll fix it